syntax = "proto3";
package Config;

message GetConfigToDtu {
  int32 offset = 1;
  uint32 time = 2;
}

message GetConfigFromDtu {
  int32 offset = 1;
  uint32 time = 2;
  int32 lock_password = 3;
  int32 lock_time = 4;
  int32 limit_power_mypower = 5;
  int32 zero_export_433_addr = 6;
  int32 zero_export_enable = 7;
  int32 netmode_select = 8;
  int32 channel_select = 9;
  int32 server_send_time = 10;
  int32 wifi_rssi = 11;
  int32 serverport = 12;
  string apn_set = 13;
  string meter_kind = 14;
  string meter_interface = 15;
  string wifi_ssid = 16;
  string wifi_passward = 17;
  string server_domain_name = 18;
  int32 inv_type = 19;
  string dtu_sn = 20;
  int32 access_model = 21;
  int32 mac_0 = 22;
  int32 mac_1 = 23;
  int32 mac_2 = 24;
  int32 mac_3 = 25;
  int32 dhcp_switch = 26;
  int32 ip_addr_0 = 27;
  int32 ip_addr_1 = 28;
  int32 ip_addr_2 = 29;
  int32 ip_addr_3 = 30;
  int32 subnet_mask_0 = 31;
  int32 subnet_mask_1 = 32;
  int32 subnet_mask_2 = 33;
  int32 subnet_mask_3 = 34;
  int32 default_gateway_0 = 35;
  int32 default_gateway_1 = 36;
  int32 default_gateway_2 = 37;
  int32 default_gateway_3 = 38;
  string ka_count = 39;
  string apn_name = 40;
  string apn_passward = 41;
  int32 sub1g_sweep_switch = 42;
  int32 sub1g_work_channel = 43;
  int32 cable_dns_0 = 44;
  int32 cable_dns_1 = 45;
  int32 cable_dns_2 = 46;
  int32 cable_dns_3 = 47;
  int32 wifi_ip_addr_0 = 48;
  int32 wifi_ip_addr_1 = 49;
  int32 wifi_ip_addr_2 = 50;
  int32 wifi_ip_addr_3 = 51;
  int32 mac_4 = 52;
  int32 mac_5 = 53;
  int32 wifi_mac_0 = 54;
  int32 wifi_mac_1 = 55;
  int32 wifi_mac_2 = 56;
  int32 wifi_mac_3 = 57;
  int32 wifi_mac_4 = 58;
  int32 wifi_mac_5 = 59;
  string gprs_imei = 60;
  string dtu_ap_ssid = 61;
  string dtu_ap_pass = 62;
}

message SetConfigToDtu {
  int32 offset = 1;
  uint32 time = 2;
  int32 lock_password = 3;
  int32 lock_time = 4;
  int32 limit_power_mypower = 5;
  int32 zero_export_433_addr = 6;
  int32 zero_export_enable = 7;
  int32 netmode_select = 8;
  int32 channel_select = 9;
  int32 server_send_time = 10;
  int32 serverport = 11;
  string apn_set = 12;
  string meter_kind = 13;
  string meter_interface = 14;
  string wifi_ssid = 15;
  string wifi_passward = 16;
  string server_domain_name = 17;
  int32 inv_type = 18;
  string dtu_sn = 19;
  int32 access_model = 20;
  int32 mac_0 = 21;
  int32 mac_1 = 22;
  int32 mac_2 = 23;
  int32 mac_3 = 24;
  int32 dhcp_switch = 25;
  int32 ip_addr_0 = 26;
  int32 ip_addr_1 = 27;
  int32 ip_addr_2 = 28;
  int32 ip_addr_3 = 29;
  int32 subnet_mask_0 = 30;
  int32 subnet_mask_1 = 31;
  int32 subnet_mask_2 = 32;
  int32 subnet_mask_3 = 33;
  int32 default_gateway_0 = 34;
  int32 default_gateway_1 = 35;
  int32 default_gateway_2 = 36;
  int32 default_gateway_3 = 37;
  string apn_name = 38;
  string apn_passward = 39;
  int32 sub1g_sweep_switch = 40;
  int32 sub1g_work_channel = 41;
  int32 cable_dns_0 = 42;
  int32 cable_dns_1 = 43;
  int32 cable_dns_2 = 44;
  int32 cable_dns_3 = 45;
  int32 mac_4 = 46;
  int32 mac_5 = 47;
  string dtu_ap_ssid = 48;
  string dtu_ap_pass = 49;
  int32 app_page = 50;
}

message SetConfigFromDtu {
  int32 offset = 1;
  uint32 time = 2;
  int32 err_code = 3;
}
